packages/core/src/scope.ts

```diff
@@ -499,6 +509,36 @@ class ScopeClass implements ScopeInterface {
     return this._propagationContext;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public getFlags(): FeatureFlag[] {
```

Comment:
If you move get and insert methods to a dedicated "buffer" type (similar to the python implementation) we can avoid polluting the scope namespace.

Reply:
I felt like the logic was short enough to include in this class, it's only really the insert method. Maybe we can ask JS SDK folks what they think?

For now, while developing/testing, I didn't want to include another package we have to link locally. But in the final iteration I'm open to making this type in `@sentry/utils`.

Reply:
discussed this w/Billy, will add a PR comment below